src/main.c

```diff
@@ -932,6 +931,9 @@ static void cmd_loop(server_rec *server, conn_t *c) {
  
       pr_cmd_dispatch(cmd);
       destroy_pool(cmd->pool);
+      session.curr_cmd = NULL;
+      session.curr_cmd_id = 0;
+      session.curr_cmd_rec = NULL;
```

Comment:
Here we ensure that we clear out any pointers to the now-dispatched command, for cleanliness (and reuse prevention).